### DISCUSSION TITLE
A pattern for users tracking the state of things they own - is this feasible in budibase

### DISCUSSION THREAD

**Discussion by pigeonflight** [Budibase/budibase#3914] [Help] [answered]
# Pattern - tracking the state of things owned
The important thing here is that a user can only get information about the things they own.

## Sample use-case:
- A service ships boxes of fixed dimension weekly between two cities. Users of the system are able to "own" one or more boxes with a weight limit of 100 kg. 
- A user may send anything that can fit in their box on a weekly basis. 
- A packer receives the goods and puts them in the boxes, if the weight exceeds 100 kg the packer must apply a penalty.
- The user is alerted about the state (weights and penalties) of their box whenever goods are loaded in their box

This is not a comprehensive use-case but it is enough to demonstrate the pattern I'm thinking about. Is this a pattern that is feasible in budibase? I'd appreciate some hints or pointers on implementing such a pattern.

**Comment by mjashanks** [comment, marked as answer]
Hey @pigeonflight 

I can't think of any reason why you would not be able to product this app with Budibase.

I would set up your app up as follows
A `Boxes` table, with 
- an `Owner` relationship (to the users table): `One User has Many Boxes` 
-  a weight column (number)

Setup an automation that sends an email to the Owner, when the Box weight exceeds 100.
